### PR TITLE
[WIP] Storage in Image Management

### PR DESCRIPTION
**Pull request by runcom**
This is tackling image management operations from https://github.com/kubernetes-incubator/ocid/issues/8.
This work is based on https://github.com/containers/image/pull/63 by vendoring @nalind's branch `add-storage-transport` here and by applying and modifying some of his patches from his ocid branch at https://github.com/nalind/ocid/tree/storage-tests.

The last commits add images pull and delete functionalities like so:

``` sh
$ sudo ./ocic images pull kubernetes/pause
$ sudo ./ocic images list
INFO[0000] id:"93a58c47ea25244e199de35f752ed7f91c25ce96947792d51fad48e0b26a2fb3" repo_tags:"kubernetes/pause:latest"
$ sudo ./ocic images delete kubernetes/pause:latest
$ sudo ./ocic images list
$ # empty list
```

Plan is:
- have containers/image#63 merged asap
- revendor containers/image here (first commit)
- add the remaining ocic commands for image management
- get this PR in

This might not be perfect as a whole when taking into account containers rootfs management but it's a start - at least in the direction of image management. Containers' rootfs management will come as part of another PR since this is already huge.

/cc @nalind @mrunalp PTAL

@nalind if you want to carry this please do so - I opened it so we have a WIP branch everyone can test and look around. Also opened this to avoid you re-vendoring everything and rebasing.

### CODE REVIEW

File: cmd/client/main.go [hunk 1]
```diff
@@ -250,6 +252,36 @@ func main() {
...
 }
 
+var listImagesCommand = cli.Command{
+	Name:  "listimages",
+	Usage: "list images",
+	Action: func(context *cli.Context) error {
+		// Set up a connection to the server.
+		conn, err := getClientConnection()
+		if err != nil {
+			return fmt.Errorf("Failed to connect: %v", err)
```
Comment (mrunalp) [inline cmd/client/main.go]: nit: Failed -> failed. We probably need to fix this all over the code.

Comment (runcom) [inline cmd/client/main.go]: I will, probably going to submit a separate PR

**Comment by mrunalp** [comment]
Nice! I will test and review in more detail.

**Comment by runcom** [comment]
@mrunalp great, I haven't tested everything yet on the image management side, will probably do so tomorrow or Monday as I implement the remaining client calls related to images.

**Comment by runcom** [comment]
added image list and delete functionalities to the cli

**Comment by rhatdan** [comment]
Any movement on this?

**Comment by runcom** [comment]
@rhatdan this will surely need a rebase now that we use another vendoring tool - I think @nalind has been working on this.

**Comment by runcom** [comment]
@nalind is gonna take this and re-open it. Closing.

**Pull request closed without merging**